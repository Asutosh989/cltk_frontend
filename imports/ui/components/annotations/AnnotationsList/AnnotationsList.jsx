import IconButton from 'material-ui/IconButton';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import Annotations from '/imports/api/collections/annotations';

AnnotationsList = React.createClass({

	propTypes: {
		text: React.PropTypes.object.isRequired,
	},

	mixins: [ReactMeteorData],

	getInitialState() {
		return {
			sortMethod: 'votes',
		};
	},

	getMeteorData() {
		let annotations = [];

		let sort = {};
		switch (this.state.sortMethod) {
			case 'votes':
				sort = { votes: -1, updated: -1 };
				break;
			case 'recent':
				sort = { updated: -1, votes: -1 };
				break;
		}

		const query = { textNode: this.props.text._id._str };

		const handle = Meteor.subscribe('annotations', query);
		annotations = Annotations.find(query, { sort }).fetch();

		return {
			annotations,
		};
	},

	addDiscussionComment() {
		const content = $(this.newCommentForm).find('textarea').val();

		Meteor.call('annotations.insert', {
			content,
			textNode: this.props.text._id._str,
		});

		$(this.newCommentForm).find('textarea').val('');
	},

	sortMethodSelect(value) {
		this.setState({
			sortMethod: value,
		})
	},

	render() {
		const currentUser = Meteor.user();

		let discussionWrapClass = 'discussion-wrap';

		if (this.state.discussionVisible) {
			discussionWrapClass += ' discussion-visible';
		}

		let textareaPlaceholder = '';
		if (currentUser) {
			textareaPlaceholder = 'Enter your comment here . . .';
		} else {
			textareaPlaceholder = 'Please login to enter a comment.';
		}

		const sortSelectedLabelStyle = {
			color: '#FFFFFF',
		};

		return (
			<div className={discussionWrapClass}>
				<div className="discussion-thread">
					<div className="add-comment-wrap paper-shadow ">
						<form
							ref={(component) => { this.newCommentForm = component; }}
							className="new-comment-form"
							name="new-comment-form"
						>
							<div className="add-comment-row-1">
								<div className="profile-picture paper-shadow">
									<img
										src={currentUser && currentUser.avatar ?
											currentUser.avatar.url : '/images/default_user.jpg'}
										alt="User"
									/>
								</div>
								<textarea
									className="new-comment-text"
									name="newCommentText"
									placeholder={textareaPlaceholder}
								/>
							</div>
							<div className="add-comment-row-2 add-comment-row">
								<div className="error-message">
									<span className="error-message-text">Please enter your text to submit.</span>
								</div>
								{ currentUser ?
									<RaisedButton
										label="Submit"
										className="submit-comment-button paper-shadow"
										onClick={this.addDiscussionComment}
									/>
									:
									<div
										className="new-comment-login"
									>
										<FlatButton
											label="Join"
											className="join-link"
											href="/sign-up"
										/>
										<FlatButton
											label="Login"
											className="login-link"
											href="/sign-in"
										/>
									</div>
								}
							</div>
						</form>
					</div>
					<div
						className="sort-by-wrap"
					>
						{/*
						 <span className="sort-by-label">Sort by:</span>
						 <RaisedButton
						 label="Top"
						 className="sort-by-option selected-sort sort-by-top"
						 onClick={this.toggleSort}>
						 </RaisedButton>
						 <RaisedButton
						 label="Newest"
						 className="sort-by-option sort-by-new"
						 onClick={this.toggleSort}>
						 </RaisedButton>
						 */}
					</div>
					{this.data.annotations.length === 0 ?
						<div className="no-results-wrap">
							<span className="no-results-text">No annotations.</span>
						</div>
						: ''
					}
					<div className="sort-method-select">
						<FlatButton
							label="Top votes"
							labelStyle={this.state.sortMethod === 'votes' ? sortSelectedLabelStyle : {}}
							backgroundColor={this.state.sortMethod === 'votes' ? '#795548' : ''}
							onClick={this.sortMethodSelect.bind(null, 'votes')}
						/>
						<FlatButton
							label="Recent"
							labelStyle={this.state.sortMethod === 'recent' ? sortSelectedLabelStyle : {}}
							backgroundColor={this.state.sortMethod === 'recent' ? '#795548' : ''}
							onClick={this.sortMethodSelect.bind(null, 'recent')}
						/>
					</div>
					{this.data.annotations.map((annotation, i) =>
						<AnnotationItem
							key={i}
							className="discussion-comment paper-shadow"
							annotation={annotation}
							currentUser={currentUser}
						/>
					)}
				</div>
			</div>
		);
	},
});
