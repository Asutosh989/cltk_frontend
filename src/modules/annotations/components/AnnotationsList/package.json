{
  "name": "AnnotationsList",
  "version": "0.0.1",
  "private": true,
  "main": "./AnnotationsList.js"
}
