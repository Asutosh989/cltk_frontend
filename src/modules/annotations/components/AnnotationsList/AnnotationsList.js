import React from 'react';
import PropTypes from 'prop-types';
import autoBind from 'react-autobind';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';


import './AnnotationsList.css';


class AnnotationsList extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
			sortMethod: 'votes',
		};

		autoBind(this);
	}

	addDiscussionComment() {
		$(this.newCommentForm).find('textarea').val('');
	}

	sortMethodSelect(value) {
		this.setState({
			sortMethod: value,
		})
	}

	render() {
		const { annotations } = this.props;

		let discussionWrapClass = 'discussion-wrap';

		if (this.state.discussionVisible) {
			discussionWrapClass += ' discussion-visible';
		}

		let textareaPlaceholder = '';
		if (currentUser) {
			textareaPlaceholder = 'Enter your comment here . . .';
		} else {
			textareaPlaceholder = 'Please login to enter a comment.';
		}

		const sortSelectedLabelStyle = {
			color: '#FFFFFF',
		};

		return (
			<div className={discussionWrapClass}>
				<div className="discussion-thread">
					<div className="add-comment-wrap paper-shadow ">
						<form
							ref={(component) => { this.newCommentForm = component; }}
							className="new-comment-form"
							name="new-comment-form"
						>
							<div className="add-comment-row-1">
								<div className="profile-picture paper-shadow">
									<img
										src={currentUser && currentUser.avatar ?
											currentUser.avatar.url : '/images/default_user.jpg'}
										alt="User"
									/>
								</div>
								<textarea
									className="new-comment-text"
									name="newCommentText"
									placeholder={textareaPlaceholder}
								/>
							</div>
							<div className="add-comment-row-2 add-comment-row">
								<div className="error-message">
									<span className="error-message-text">Please enter your text to submit.</span>
								</div>
								{ currentUser ?
									<RaisedButton
										label="Submit"
										className="submit-comment-button paper-shadow"
										onClick={this.addDiscussionComment}
									/>
									:
									<div
										className="new-comment-login"
									>
										<FlatButton
											label="Join"
											className="join-link"
											href="/sign-up"
										/>
										<FlatButton
											label="Login"
											className="login-link"
											href="/sign-in"
										/>
									</div>
								}
							</div>
						</form>
					</div>
					<div
						className="sort-by-wrap"
					>
						{/*
						 <span className="sort-by-label">Sort by:</span>
						 <RaisedButton
							 label="Top"
							 className="sort-by-option selected-sort sort-by-top"
							 onClick={this.toggleSort}>
						 </RaisedButton>
						 <RaisedButton
							 label="Newest"
							 className="sort-by-option sort-by-new"
							 onClick={this.toggleSort}>
						 </RaisedButton>
						 */}
					</div>
					{annotations && annotations.length === 0 ?
						<div className="no-results-wrap">
							<span className="no-results-text">No annotations.</span>
						</div>
						: ''
					}
					<div className="sort-method-select">
						<FlatButton
							label="Top votes"
							labelStyle={this.state.sortMethod === 'votes' ? sortSelectedLabelStyle : {}}
							backgroundColor={this.state.sortMethod === 'votes' ? '#795548' : ''}
							onClick={this.sortMethodSelect.bind(null, 'votes')}
						/>
						<FlatButton
							label="Recent"
							labelStyle={this.state.sortMethod === 'recent' ? sortSelectedLabelStyle : {}}
							backgroundColor={this.state.sortMethod === 'recent' ? '#795548' : ''}
							onClick={this.sortMethodSelect.bind(null, 'recent')}
						/>
					</div>
					{annotations && annotations.map((annotation, i) =>
						<AnnotationItem
							key={i}
							className="discussion-comment paper-shadow"
							annotation={annotation}
							currentUser={currentUser}
						/>
					)}
				</div>
			</div>
		);
	}
}


AnnotationsList.propTypes = {
	text: PropTypes.object.isRequired,
};

export default AnnotationsList;
