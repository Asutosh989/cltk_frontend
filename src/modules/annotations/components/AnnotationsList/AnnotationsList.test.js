import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import AnnotationsList from './AnnotationsList';

describe('AnnotationsList', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<AnnotationsList
						text={{
							_id: {
								_str: 'testId',
							},
						}}
					/>
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
