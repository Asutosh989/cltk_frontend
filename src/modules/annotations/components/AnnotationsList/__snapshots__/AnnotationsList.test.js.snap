// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnotationsList renders correctly 1`] = `
<div
  className="discussion-wrap"
>
  <div
    className="discussion-thread"
  >
    <div
      className="add-comment-wrap paper-shadow "
    >
      <form
        className="new-comment-form"
        name="new-comment-form"
      >
        <div
          className="add-comment-row-1"
        >
          <div
            className="profile-picture paper-shadow"
          >
            <img
              alt="User"
              src="/images/default_user.jpg"
            />
          </div>
          <textarea
            className="new-comment-text"
            name="newCommentText"
            placeholder="Please login to enter a comment."
          />
        </div>
        <div
          className="add-comment-row-2 add-comment-row"
        >
          <div
            className="error-message"
          >
            <span
              className="error-message-text"
            >
              Please enter your text to submit.
            </span>
          </div>
          <div
            className="new-comment-login"
          >
            <a
              className="join-link"
              disabled={false}
              href="/sign-up"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "rgba(0, 0, 0, 0)",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "minWidth": 88,
                  "muiPrepared": true,
                  "outline": "none",
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                  "zIndex": 1,
                }
              }
              tabIndex={0}
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <span
                  style={
                    Object {
                      "fontSize": 14,
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "muiPrepared": true,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "verticalAlign": "middle",
                    }
                  }
                >
                  Join
                </span>
              </div>
            </a>
            <a
              className="login-link"
              disabled={false}
              href="/sign-in"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              onTouchTap={[Function]}
              style={
                Object {
                  "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                  "backgroundColor": "rgba(0, 0, 0, 0)",
                  "border": 10,
                  "borderRadius": 2,
                  "boxSizing": "border-box",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": "inherit",
                  "fontWeight": "inherit",
                  "height": 36,
                  "lineHeight": "36px",
                  "margin": 0,
                  "minWidth": 88,
                  "muiPrepared": true,
                  "outline": "none",
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "verticalAlign": "middle",
                  "zIndex": 1,
                }
              }
              tabIndex={0}
            >
              <div
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
              >
                <span
                  style={
                    Object {
                      "fontSize": 14,
                      "fontWeight": 500,
                      "letterSpacing": 0,
                      "muiPrepared": true,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "position": "relative",
                      "textTransform": "uppercase",
                      "verticalAlign": "middle",
                    }
                  }
                >
                  Login
                </span>
              </div>
            </a>
          </div>
        </div>
      </form>
    </div>
    <div
      className="sort-by-wrap"
    />
    
    <div
      className="sort-method-select"
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#795548",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Top votes
          </span>
        </div>
      </button>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Recent
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
